roles/rhsm-repos/tasks/main.yaml

```diff
@@ -0,0 +1,38 @@
+---
+- name: disable unneeded repos
+  command: subscription-manager repos --disable='*'
```

Comment:
There should probably be a prior step to determine what repos are currently enabled and only disable/enable when needed. Examples below shows what I mean as well as moving the conditional around:

```
- name: Example playbooks
  hosts: example_hosts
  roles:
  - role: rhsm-repos
    when: "{{ ansible_distribution == 'RedHat' and openshift_manage_repos is true }}"
```
```
# a bit handwavy, since this output needs to be filtered, or a diff command used, but this example assumes that the output here is just a bare list of repos enabled
- command: subscription-manager repos --list-enabled
  register: subscription_manager_repos
  changed_when: false

- block:
    - name: disable repos
      command: "subscription-manager{{ ' --disable='.join(repos_to_remove) + ' --enable'.join(repos_to_add)}}"
  when: "{{ repos_to_remove | length > 0 or repos_to_add | length > 0 }}"
  vars:
    repo_list:
    - rhel-7-server-rpms
    - rhel-7-server-extras-rpms
    - rhel-7-server-ose-3.3-rpms
    repos_to_remove: "{{ subscription_manager_repos | difference(repo_list) }}"
    repos_to_add: "{{ repo_list | difference(subscription_manager_repos) }}"

....
```